pyinfra/api/operation.py

```diff
@@ -222,30 +219,9 @@ def decorated_func(*args, **kwargs):
         host.current_op_global_kwargs = None
 
         # Add host-specific operation data to state
-        #
-
-        # We're doing some commands, meta/ops++
-        state.meta[host]["ops"] += 1
-        state.meta[host]["commands"] += len(commands)
-
-        if commands:
-            state.meta[host]["ops_change"] += 1
-        else:
-            state.meta[host]["ops_no_change"] += 1
-
-        operation_meta = OperationMeta(op_hash, commands)
-
-        # Add the server-relevant commands
-        op_data = {
-            "commands": commands,
-            "global_kwargs": global_kwargs,
-            "operation_meta": operation_meta,
-        }
-        state.set_op_data(host, op_hash, op_data)
-
-        # If we're already in the execution phase, execute this operation immediately
-        if state.is_executing:
-            _execute_immediately(state, host, op_data, op_meta, op_hash)
+        state, operation_meta = _update_state_meta(
```

Comment:
This needn't return `state` because it mutates it, which I think we should also add a commend indicating.

Reply:
https://github.com/Fizzadar/pyinfra/pull/883/commits/f30aa877f7cdb5349d0f41f3459886c3255532af